tests/wicked/before_test.pm

```diff
@@ -138,6 +138,13 @@ sub run {
             zypper_ar($repo_url . generate_version('_') . '/', name => 'wicked_maintainers', no_gpg_check => 1, priority => 60);
             $package_list .= ' ndisc6';
         }
+        if (check_var('WICKED', 'startandstop')) {
+            # No firewalld on sles 12-SP5 (bsc#1180116)
+            if (!is_sle('<=12-SP5')) {
+                zypper_call('-q in firewalld', timeout => 400);
```

Comment:
hm , AFAIK firewalld suppose to be installed by default on all SLE 15

Reply:
yes you messing up here with https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15709/files#diff-4444ed2221a3fad7639c99baa4ed295548f6638dd2cda4daf842a1307a7e536cR39

Reply:
don't understand the link!

We use autoyast to install and it looks like there is no firewalld by default see
 * tumbelweed: http://openqa.wicked.suse.de/tests/84114/logfile?filename=serial_terminal.txt
 * sle-15-sp4: http://openqa.wicked.suse.de/tests/84117/logfile?filename=serial_terminal.txt

Reply:
just look into `before_test.pm` in line 39 . the point is that when you getting into line 144 there is already logic which trying to handle firewalld . so at least there is no point to try to stop it because if it is installed it is already stopped

Reply:
maybe you can this install request into module itself ? because currently you installed firewalld even for `wicked_version < 0.6.70`  and module will not run after all

Reply:
Now I get it!

Or we simply install firewalld via autoyast and deactivate it always?

I mean what we need is firewall deactivated for all tests. But installing it during tests is hard, as we would need to setup network and we have that install logic only in `before` test.

Reply:
isn't it will fail if package is already installed ?

Reply:
> sle-15-sp4: http://openqa.wicked.suse.de/tests/84117/logfile?filename=serial_terminal.txt

this is NOT SLE 15 SP4 but Leap !  For SLE 15 SP* firewalld is default firewall which should be present by default and if it is not presented it is a bug and blindly installing it is hiding a bug . Regarding Leap / TW yes you might need to install it but you need to use proper `is_leap` expression , regarding TW it will be more complicated because there is no `is_tw` the only way to figure out that it TW is check that it is NOT something else which would be ugly

Reply:
Sorry for not pasting the right link:
  * SLE 15-SP4 http://openqa.wicked.suse.de/tests/84118/logfile?filename=serial_terminal.txt
  * all my tests: http://openqa.wicked.suse.de/tests/overview?build=firewall_ext
But it looks like TW and LEAP.

Also installing something which is already there, do not produce a errcode:
```bash
> zypper search firewalld
Loading repository data...
Reading installed packages...

S  | Name                        | Summary                                                             | Type
---+-----------------------------+---------------------------------------------------------------------+-----------
i+ | firewalld                   | A firewall daemon with D-Bus interface providing a dynamic firewall | package
   | firewalld                   | A firewall daemon with D-Bus interface providing a dynamic firewall | srcpackage
i  | firewalld-lang              | Translations for package firewalld                                  | package
   | firewalld-prometheus-config | Firewalld configuration file for Prometheus                         | package
   | firewalld-rpcbind-helper    | Tool for static port assignment of NFSv3, ypserv, ypbind services   | package
   | keylime-firewalld           | Firewalld service file for keylime                                  | package
   | susefirewall2-to-firewalld  | Basic SuSEfirewall2 to FirewallD migration script                   | package

> sudo zypper -n -q in firewalld
[sudo] password for root: 

> echo $?
0
```

If it hide a error, when we blindly install the package ... yes it does. But I think this is not the focus of these tests?!

Reply:
ok , I found what I missed initially . Firewalld will be installed by default when you using default installation path . But this test runs against something created by autoyast profile anyway so it is kinda expected that we don't have firewalld here